homeassistant/components/manual/alarm_control_panel.py

```diff
@@ -318,7 +318,10 @@ def alarm_trigger(self, code=None):
         No code needed, a trigger time of zero for the current state
         disables the alarm.
         """
-        if not self._trigger_time_by_state[self._active_state]:
+        try:
+            if not self._trigger_time_by_state[self._active_state]:
+                return
+        except KeyError:
```

Comment:
I think we should have a separate check for triggered state and return if that is true. That makes it more clear.

Reply:
ok, I changed that but I'm afraid I wouldn't know where to start with respect to the test. Can you give me a link to a similar test case?

Reply:
Look at the existing tests for the manual alarm_control_panel.

https://github.com/home-assistant/home-assistant/blob/8de84c53a14f42e5ebc9e231cd60616ca4d73197/tests/components/manual/test_alarm_control_panel.py#L83-L119

Make sure the state is triggered and then call the service to trigger again.